R/getMadratGraph.R

```diff
@@ -1,102 +1,103 @@
 #' getMadratGraph
-#' 
-#' Function returns the madrat graph of all linkages of full, calc, and read functions of the given madrat 
+#'
+#' Function returns the madrat graph of all linkages of full, calc, and read functions of the given madrat
 #' based packages. Linkages to subfunctions of read functions (i.e. download, correct or convert functions)
 #' are not listed separately, but collectively referred to through the corresponding read function.
-#' 
-#' 
+#'
+#'
 #' @param packages A character vector with packages for which the available Sources/Calculations should be returned
 #' @param globalenv Boolean deciding whether sources/calculations in the global environment should be included or not
-#' @return A data frame with 4 columns: from (source function), from_package (package the source function originates from),
-#' to (function which is using the source), to_package (package of the using function)
+#' @return A data frame with 4 columns: from (source function), from_package (package the source function originates
+#' from), to (function which is using the source), to_package (package of the using function)
 #' @author Jan Philipp Dietrich
 #' @seealso \code{\link{getCalculations}}, \code{\link{getConfig}}
 #' @importFrom stringi stri_match_all_regex stri_extract_all
 #' @export
 
-getMadratGraph <- function(packages=installedMadratUniverse(), globalenv=getConfig("globalenv")) {
-  
-  if (is.null(getOption("MadratCache"))) options(MadratCache = new.env(size = NA))
-  
+
+getMadratGraph <- function(packages = installedMadratUniverse(), globalenv = getConfig("globalenv")) {
+
+  if (is.null(getOption("MadratCache"))) options(MadratCache = new.env(size = NA)) # nolint
```

Comment:
Just pointing out how this would look with `withr`:
```suggestion
  if (is.null(getOption("MadratCache"))) withr::local_options(c(MadratCache = new.env(size = NA)))
```
This will of course only change the option until the end of a getMadratGraph call. If we want to set it globally for the entire session we should leave it as it is.

Reply:
must be set globally as it does not make sense to create a cache which is immediately deleted again afterwards